corehq/apps/settings/forms.py

```diff
@@ -181,7 +181,18 @@ def __init__(self, *, allow_phone_2fa, **kwargs):
 
 
 class HQTOTPDeviceForm(TOTPDeviceForm):
-    token = forms.IntegerField(required=False, label=_("Token"), min_value=1, max_value=int('9' * totp_digits()))
+    token = forms.IntegerField(
+        required=False,
+        label=_("Token"),
+        min_value=1,
+        max_value=int('9' * totp_digits()),
+        widget=forms.TextInput(
+            attrs={
+                'pattern': r'^[0-9]{6}$',
+                'maxlength': totp_digits(),
```

Comment:
Should this also set `minlength` (is that a thing)? Otherwise it seems like it may be possible to enter a shorter matching code if the code has leading zeros.

Reply:
The `pattern` attribute here will validate that it is exactly six digits and won't allow the form to be submitted otherwise. The `maxlength` actually keeps someone from typing too many characters into the field. `minlength` is a valid attribute, but it's not useful here since it's just a validator, which is already covered by the `pattern`.

It didn't seem to me like we were concerned with mobile devices on this form. If that's a concern, we can set `'inputmode': 'numeric'`. It probably doesn't hurt anything to just do that anyway. https://github.com/dimagi/commcare-hq/pull/35910/commits/777a921cc9646b7f7b735512ed695b5e7a49a25d

Reply:
Oh, yes, of course. Thanks for pointing that out, and I need to read more carefully.